tools/kit_builder.py

```
@@ -82,6 +104,31 @@ def degree_list(values, what):
     return values
 
 
+def validate_sample(kit_dir, source, what):
+    """The WAV a sample pad names must be a plain file in the kit folder (no path, no
+    link) and 16-bit 48 kHz mono PCM of 1 to MAX_SAMPLE_FRAMES frames, so sound/ can
+    play it as it is."""
+    if not isinstance(source, str) or source in ("", ".", "..") or os.path.basename(source) != source:
+        raise KitError(f"{what}: source must be a file name inside the kit folder, got {source!r}")
+    path = os.path.join(kit_dir, source)
+    if os.path.islink(path):
+        raise KitError(f"{what}: {source} is a link; a kit sample is a file inside the kit folder")
```

Ruff (0.16.3)

[warning] Avoid specifying long messages outside the exception class

(TRY003)